homeassistant/components/automation/event.py

```diff
@@ -28,11 +29,12 @@ def trigger(hass, config, action):
     event_type = config.get(CONF_EVENT_TYPE)
     event_data = config.get(CONF_EVENT_DATA)
 
+    @asyncio.coroutine
     def handle_event(event):
         """Listen for events and calls the action when data matches."""
         if not event_data or all(val == event.data.get(key) for key, val
                                  in event_data.items()):
-            action({
+            hass.add_job(action, {
```

Comment:
Might want to do a iscoroutinefunction check on action and schedule it on the loop if its a coroutine, that way coroutine action's could stay on the event loop.